### PR TITLE
Move the /rows endpoint to its own service

### PR DESCRIPTION
**Pull request by severo**
We create a new service, services/rows, which handles the /rows endpoint. services/api now serves the rest of the endpoints, but not /rows.

### CODE REVIEW

File: services/rows/src/rows/app.py [hunk 1]
```diff
@@ -0,0 +1,101 @@
+# SPDX-License-Identifier: Apache-2.0
+# Copyright 2022 The HuggingFace Authors.
```
Comment (AndreaFrancis) [inline services/rows/src/rows/app.py:2]:
```suggestion
# Copyright 2023 The HuggingFace Authors.
```

File: chart/env/staging.yaml [hunk 2]
```diff
@@ -211,6 +216,20 @@ api:
...
+rows:
+  uvicornNumWorkers: "1"
+
+  replicas: 1
+  service:
+    type: NodePort
+  resources:
+    requests:
+      cpu: 100m
+      memory: "512Mi"
```
Comment (lhoestq) [inline chart/env/staging.yaml:228]: we'll need to increase that number a bit - see (internal) https://huggingface.slack.com/archives/C04L6P8KNQ5/p1688484608969739

File: services/rows/src/rows/config.py [hunk 3]
```diff
@@ -0,0 +1,41 @@
+# SPDX-License-Identifier: Apache-2.0
+# Copyright 2022 The HuggingFace Authors.
```
Comment (AndreaFrancis) [inline services/rows/src/rows/config.py:2]:
```suggestion
# Copyright 2023 The HuggingFace Authors.
```

**Review by AndreaFrancis — APPROVED** [review]
Awesome!

File: services/reverse-proxy/README.md [hunk 4]
```diff
@@ -17,8 +17,9 @@ It takes various environment variables, all of them are mandatory:
 - `ASSETS_DIRECTORY`: the directory that contains the static assets, eg `/assets`
 - `HOST`: domain of the reverse proxy, eg `localhost`
 - `PORT`: port of the reverse proxy, eg `80`
-- `URL_ADMIN`= URL of the admin, eg `http://admin:8080`
+- `URL_ADMIN`= URL of the admin, eg `http://admin:8081`
 - `URL_API`= URL of the API, eg `http://api:8080`
+- `URL_ROWS`= URL of the rows service, eg `http://admin:8082`
```
Comment (lhoestq) [inline services/reverse-proxy/README.md:22]:
```suggestion
- `URL_ROWS`= URL of the rows service, eg `http://rows:8082`
```

File: tools/docker-compose-datasets-server.yml [hunk 5]
```diff
@@ -76,11 +105,11 @@ services:
...
       # uvicorn
       API_UVICORN_HOSTNAME: 0.0.0.0 # required for docker compose
-      API_UVICORN_NUM_WORKERS: ${API_UVICORN_NUM_WORKERS-2}
-      API_UVICORN_PORT: ${API_UVICORN_PORT-8080}
+      API_UVICORN_NUM_WORKERS: ${ROWS_UVICORN_NUM_WORKERS-2}
+      API_UVICORN_PORT: ${ROWS_UVICORN_PORT-8080}
     ports:
       # for debug
-      - ${API_UVICORN_PORT-8080}:${API_UVICORN_PORT-8080}
+      - ${ROWS_UVICORN_PORT-8080}:${ROWS_UVICORN_PORT-8080}
```
Comment (lhoestq) [inline tools/docker-compose-datasets-server.yml:112]:
```suggestion
      - ${ROWS_UVICORN_PORT-8082}:${ROWS_UVICORN_PORT-8082}
```

File: tools/docker-compose-datasets-server.yml [hunk 6]
```diff
@@ -76,11 +105,11 @@ services:
       PROMETHEUS_MULTIPROC_DIR: ${PROMETHEUS_MULTIPROC_DIR-}
       # uvicorn
       API_UVICORN_HOSTNAME: 0.0.0.0 # required for docker compose
-      API_UVICORN_NUM_WORKERS: ${API_UVICORN_NUM_WORKERS-2}
-      API_UVICORN_PORT: ${API_UVICORN_PORT-8080}
+      API_UVICORN_NUM_WORKERS: ${ROWS_UVICORN_NUM_WORKERS-2}
+      API_UVICORN_PORT: ${ROWS_UVICORN_PORT-8080}
```
Comment (lhoestq) [inline tools/docker-compose-datasets-server.yml:109]:
```suggestion
      API_UVICORN_PORT: ${ROWS_UVICORN_PORT-8082}
```

File: tools/docker-compose-dev-datasets-server.yml [hunk 7]
```diff
@@ -78,11 +107,12 @@ services:
       PROMETHEUS_MULTIPROC_DIR: ${PROMETHEUS_MULTIPROC_DIR-}
       # uvicorn
       API_UVICORN_HOSTNAME: 0.0.0.0 # required for docker compose
-      API_UVICORN_NUM_WORKERS: ${API_UVICORN_NUM_WORKERS-2}
-      API_UVICORN_PORT: ${API_UVICORN_PORT-8080}
+      API_UVICORN_NUM_WORKERS: ${ROWS_UVICORN_NUM_WORKERS-2}
+      API_UVICORN_PORT: ${ROWS_UVICORN_PORT-8080}
```
Comment (lhoestq) [inline tools/docker-compose-dev-datasets-server.yml:111]:
```suggestion
      API_UVICORN_PORT: ${ROWS_UVICORN_PORT-8082}
```

File: tools/docker-compose-dev-datasets-server.yml [hunk 8]
```diff
@@ -78,11 +107,12 @@ services:
...
     network_mode: ${DEV_NETWORK_MODE}
     ports:
-      - ${API_UVICORN_PORT-8080}:${API_UVICORN_PORT-8080}
+      # for debug
+      - ${ROWS_UVICORN_PORT-8080}:${ROWS_UVICORN_PORT-8080}
```
Comment (lhoestq) [inline tools/docker-compose-dev-datasets-server.yml:115]:
```suggestion
      - ${ROWS_UVICORN_PORT-8082}:${ROWS_UVICORN_PORT-8082}
```

**Review by lhoestq — APPROVED** [review]
Looks all good, thanks !

File: chart/env/staging.yaml (hunk 2 above)
Comment (severo) [reply chart/env/staging.yaml:228]: oh yes. Let's do it in another PR?

**Review by XciD — COMMENTED** [review]
LGTM

**Comment by severo** [comment]
> LGTM

cool! thanks for reviewing

**Pull request merged**